File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    goodNodes(root) {
        return this.rec(root, -Infinity)
    }

    rec(root, max){
        let resL = 0
        let resR = 0
        if(root.val >= max){
            
            max = root.val
            console.log("PREMAX", max)
            if(root.left) resL = this.rec(root.left, max)
            if(root.right) resR =  this.rec(root.right, max)
            console.log("CURR", root.val)
            console.log("MAX", max)
        console.log(resL + resR + 1)
            return resL + resR + 1
        }
        else{
            if(root.left) resL = this.rec(root.left, max)
            if(root.right) resR =  this.rec(root.right, max)
            console.log("CURR", root.val)
            console.log("MAX", max)
        console.log(resL + resR)
            return resL + resR
        }
        
    }
}

File: Data Structures & Algorithms/binary-tree-maximum-path-sum/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxPathSum(self, root: Optional[TreeNode]) -> int:
        #l + root + r
        #root
        self.ans = root.val
        def rec(n):
            if not n:
                return 0
            l = max(0, rec(n.left))
            r = max(0, rec(n.right))
            total = l + r + n.val
            self.ans = max(self.ans, total)
            return max(l, r) + n.val

        rec(root)
        return self.ans

            